cmd/clusterctl/client/cluster/proxy.go

```diff
@@ -142,10 +142,13 @@ func (k *proxy) GetConfig() (*rest.Config, error) {
 	}
 	restConfig, err := clientcmd.NewDefaultClientConfig(*config, configOverrides).ClientConfig()
 	if err != nil {
-		if strings.HasPrefix(err.Error(), "invalid configuration:") {
-			return nil, errors.New(strings.Replace(err.Error(), "invalid configuration:", "invalid kubeconfig file; clusterctl requires a valid kubeconfig file to connect to the management cluster:", 1))
+		if !strings.HasPrefix(err.Error(), "invalid configuration:") {
+			return nil, err
```

Comment:
any reason to not use controller runtime builtin GetConfig() which handles the fallback for you? i.e.
```
ctrconfig "sigs.k8s.io/controller-runtime/pkg/client/config"
restConfig, err := ctrconfig.GetConfig()
```

Reply:
Because GetConfig() uses a global "kubeconfig" flag which clusterctl does not set.